plugins/MLTModule.cpp

```diff
@@ -68,129 +67,158 @@ MLTModule::decode_geoid(uint64_t _geoid_int)
 void
 MLTModule::init(std::shared_ptr<appfwk::ConfigurationManager> mcfg)
 {
-  auto mtrg = mcfg->get_dal<appmodel::MLTModule>(get_name());
+  TLOG_DEBUG(TLVL_ENTER_EXIT_METHODS) << get_name() << ": Entering init() method";
+
+  m_mtrg = mcfg->get_dal<appmodel::MLTModule>(get_name());
+  // Get the session to access the detector configuration
+  m_session = mcfg->session();
+
+  TLOG_DEBUG(TLVL_ENTER_EXIT_METHODS) << get_name() << ": Exiting init() method";
+}
+
+void
+MLTModule::do_configure(const nlohmann::json& /*obj*/)
+{
+  TLOG_DEBUG(TLVL_ENTER_EXIT_METHODS) << get_name() << ": Entering conf() method";
 
   // Get the inputs
-  std::string candidate_input;
-  std::string inhibit_input;
-  for(auto con : mtrg->get_inputs()){
-    if(con->get_data_type() == datatype_to_string<dfmessages::TriggerDecision>()) {
-      m_decision_input = get_iom_receiver<dfmessages::TriggerDecision>(con->UID());
-    } else if(con->get_data_type() == datatype_to_string<dfmessages::TriggerInhibit>()) {
+  for (auto con : m_mtrg->get_inputs()) {
+    if (con->get_data_type() == datatype_to_string<dfmessages::TriggerDecision>()) {
+        m_decision_input = get_iom_receiver<dfmessages::TriggerDecision>(con->UID());
+    } else if (con->get_data_type() == datatype_to_string<dfmessages::TriggerInhibit>()) {
       m_inhibit_input = get_iom_receiver<dfmessages::TriggerInhibit>(con->UID());
     }
   }
 
   // Get the outputs
-  for(auto con : mtrg->get_outputs()){
-    if(con->get_data_type() == datatype_to_string<dfmessages::TriggerDecision>())
+  for (auto con : m_mtrg->get_outputs()) {
+    if (con->get_data_type() == datatype_to_string<dfmessages::TriggerDecision>())
       m_decision_output = get_iom_sender<dfmessages::TriggerDecision>(con->UID());
   }
 
-  // Get the session to access the detector configuration
-  auto session = mcfg->session();
-
-  hdf5libs::HDF5SourceIDHandler::source_id_geo_id_map_t geoidmap = hdf5libs::HDF5SourceIDHandler::make_source_id_geo_id_map(session);
-
+  hdf5libs::HDF5SourceIDHandler::source_id_geo_id_map_t geoidmap =
+    hdf5libs::HDF5SourceIDHandler::make_source_id_geo_id_map(m_session);
```

Comment:
That's the only place `m_session` is being used. I guess the `auto session...` wasn't ideal either, but if change it, then why not to 

```suggestion
    hdf5libs::HDF5SourceIDHandler::make_source_id_geo_id_map(m_mtrg->session());
```
so we don't end up storing that object? Otherwise we could as well store far more (like the list of inputs etc.)

Reply:
As explained before, session is not part of the configuration block for the MLTModule but a separete conf object. Therefore we need to 'stash' both the session and the module conf object.